app/audit/PSPAmendment.scala

```diff
@@ -16,9 +16,53 @@
 
 package audit
 
-case class PSPAmendment(pspId:String, email: String) extends AuditEvent {
+import play.api.libs.json.Reads._
+import play.api.libs.json._
 
-  override def auditType: String = "PSPAmendment"
-  override def details: Map[String, String] = Map("pspId" -> pspId, "emailId" -> email)
+case class PSPAmendment(
+                         pspId: String,
+                         originalSubscriptionDetails: JsValue,
+                         updatedSubscriptionDetails: JsValue
+                       ) extends AuditEvent {
+
+  override def auditType: String = "PensionSchemePractitionerAmendment"
+
+  private val original: JsObject =
+    originalSubscriptionDetails.as[JsObject]
+      .-("subscriptionType")
+
+  private val updates: JsObject =
+    updatedSubscriptionDetails.as[JsObject]
+      .-("pspId")
```

Comment:
Minor point: you can string these onto one line and not bother with .-, i.e. updatedSubscriptionDetails.as[JsObject] - "pspId" - "subscriptionType" - "areYouUKResident" - just might be a bit more readable and concise

Reply:
Yeah I like that better too, done